src/map/navmesh.cpp

```diff
@@ -509,11 +509,11 @@ bool CNavMesh::findFurthestValidPoint(const position_t& startPosition, const pos
     return true;
 }
 
-void CNavMesh::snapToValidPosition(position_t& position)
+void CNavMesh::snapToValidPosition(position_t& position, float targetY, bool force)
 {
     TracyZoneScoped;
 
-    if (!m_navMesh)
+    if (!m_navMesh || !targetY || (!force && abs(position.y - targetY) < 0.1))
```

Comment:
Does this need to be 0.1f?
Also floats are only falsy if they're 0.0f apparently

Reply:
Yes this needs to be 0.1f since vertical distance is so low. This just to make sure we are snapping the entity to the correct floor and that they won't go through the floor. Anything over 0.0f we want to make sure we snap, generally this only triggers about 4 times per tick with 30-40 mobs on the player since it only triggers on things like stairs or when a mob wants to fall through a floor. So cost even with this low threshold is miniscule but ensures quality.

Reply:
Sorry, I meant more literally does it need an f lol